scheduler/pkg/kafka/gateway/consumer.go

```diff
@@ -169,6 +174,22 @@ func collectHeaders(kheaders []kafka.Header) map[string]string {
 	return headers
 }
 
+func (kc *InferKafkaConsumer) Produce(msg *kafka.Message, deliveryChan chan kafka.Event) error {
+	kc.producerMu.RLock()
+	defer kc.producerMu.RUnlock()
+	if kc.running.Load() {
+		return kc.producer.Produce(msg, deliveryChan)
+	} else {
+		return fmt.Errorf("The infer producer %s is no longer running", kc.producer.String())
```

Comment:
:question: No `logger`?  The `kc` instance has a logger that presumably we should still be allowed to use here

Reply:
will add log message